src/sentry/integrations/slack/event_endpoint.py

```diff
@@ -3,81 +3,114 @@
 import json
 import re
 import six
+from collections import defaultdict
 
 from django.conf import settings
 
 from sentry import http
 from sentry.api.base import Endpoint
+from sentry.incidents.models import Incident
 from sentry.models import Group, Project
 
-from .utils import build_attachment, logger
 from .requests import SlackEventRequest, SlackRequestError
+from .utils import (
+    build_group_attachment,
+    build_incident_attachment,
+    logger,
+)
 
 # XXX(dcramer): this could be more tightly bound to our configured domain,
 # but slack limits what we can unfurl anyways so its probably safe
-_link_regexp = re.compile(r'^https?\://[^/]+/[^/]+/[^/]+/issues/(\d+)')
+_link_regexp = re.compile(r'^https?\://[^/]+/[^/]+/[^/]+/(issues|incidents)/(\d+)')
+
+
+def unfurl_issues(integration, issue_map):
+    results = {
+        g.id: g for g in Group.objects.filter(
+            id__in=set(issue_map.keys()),
+            project__in=Project.objects.filter(
+                organization__in=integration.organizations.all(),
+            )
+        )
+    }
+    if not results:
+        return {}
+
+    return {
+        v: build_group_attachment(results[k]) for k, v in six.iteritems(issue_map)
+        if k in results
+    }
+
+
+def unfurl_incidents(integration, incident_map):
+    results = {
+        i.id: i for i in Incident.objects.filter(
+            id__in=set(incident_map.keys()),
+            organization__in=integration.organizations.all(),
+        )
+    }
+    if not results:
+        return {}
+
+    return {
+        v: build_incident_attachment(results[k]) for k, v in six.iteritems(incident_map)
+        if k in results
+    }
 
 
 # XXX(dcramer): a lot of this is copied from sentry-plugins right now, and will
 # need refactored
 class SlackEventEndpoint(Endpoint):
+    event_handlers = {
+        'issues': unfurl_issues,
+        'incidents': unfurl_incidents,
+    }
+
     authentication_classes = ()
     permission_classes = ()
 
-    def _parse_issue_id_from_url(self, link):
+    def _parse_url(self, link):
         match = _link_regexp.match(link)
         if not match:
-            return
+            return None, None
```

Comment:
Could be nice if this returned a `namedtuple` or something. Multiple returns like this in python are always a bit confusing I think. Especially since you're just returning match groups with no context (1 and 2)

Reply:
I feel like defining a `namedtuple` for this one method is a little overkill. I settled for just adding a better comment to the method.